### PR TITLE
Add initial support to media files

### PR DESCRIPTION
**Pull request by efernandesng**
I'll complete tests and examples in coming days.

- [x] Initial implementation
- [x] Add example
- [ ] Add tests

Related issue: #1

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -121,9 +133,19 @@ export default function(deckName) {
...
 
     zip.file('collection.anki2', new Buffer(binaryArray));
-    zip.file('media', '{}');
+    zip.file('media', JSON.stringify(mediaObj));
+
+    if(media.length > 0) {
+      for (let i = 0; i < media.length; i++) {
+        zip.file(i, media[i].data);
+      }
+    }
```
Comment (ewnd9) [inline src/index.js]: Could you replace it with 

```js
media.forEach((item, i) => {
  zip.file(i, item.data);
})
```

for consistency, please?

Comment (efernandesng) [inline src/index.js]: done :-)

I simplified and do it in a single line.

```javascript
media.forEach((item, i) => zip.file(i, item.data));
```

**Comment by ewnd9** [comment]
@efernandesng Do you need any help?

**Comment by efernandesng** [comment]
Sorry.. I am with lack of time. I'll try finish next week :)

**Comment by efernandesng** [comment]
@ewnd9, you have some solution to compare zip files in tests?

**Comment by ewnd9** [comment]
@efernandesng I guess you could simply extract content of the zip files to `/tmp` and compare the files itself (except `sql` file, as I noted in https://github.com/ewnd9/anki-apkg-export/blob/811ef1864a27bcec1468ec0ed69b554167cf2e04/test/test.js#L24-L27 there are random numbers involved).

---

About the browser example: 
`buffer-loader` is neat, but doesn't make sense to me right now. I think a user of the library would expect to add media files from local fs via `<input type="file" />` or via presenting a URL from an external resource which will be downloaded in browser via ajax.

**Comment by efernandesng** [comment]
You are right about `buffer-loader`, but it was only to follow the previous logic.. I think that form example would be nice :)

To use a form, maybe we need use [File](https://developer.mozilla.org/en-US/docs/Web/API/File) and [FileReader](https://developer.mozilla.org/en-US/docs/Web/API/FileReader) and then create a Buffer object.

With external URL, maybe we will have problems with cross-domain requests.

**Comment by ewnd9** [comment]
Yep, I've checked `FileReader` out of curiosity and it works like a charm:

```js
'use strict';

const JSZip = require('jszip');
const filesaver = require('filesaver.js');

const input = document.getElementById('input');

input.onchange = function(e) {
  const reader = new FileReader();
  const filename = e.target.files[0].name;

  reader.onload = function(e) {
    const zip = new JSZip();
    zip.file(filename, e.target.result);

    zip
      .generateAsync({
        type: 'blob',
        mimeType: 'application/ods',
        compression: "DEFLATE"
      })
      .then(function(file) {
        filesaver.saveAs(new Blob([file], {
          type: 'application/octet-stream'
        }), 'test.zip');
      })
      .catch(err => console.log(err.stack || err));
  };

  reader.readAsArrayBuffer(e.target.files[0]);
};
```

And the browser part is possible with `fetch`:

```js
'use strict';

const JSZip = require('jszip');
const filesaver = require('filesaver.js');

const myInit = { 
  method: 'GET'
};

fetch('https://raw.githubusercontent.com/ewnd9/anki-apkg-export/39ebdd664ab23b5237eee95b7dd88c457e263a20/example/assets/anki.png',myInit)
  .then(function(response) {
    return response.blob();
  })
  .then(function(myBlob) {
    const zip = new JSZip();
    zip.file('test.png', myBlob);
    console.log(myBlob);
    
    zip
      .generateAsync({
        type: "blob",
        mimeType: "application/ods",
        compression: "DEFLATE"
      })
      .then(function(file) {
        filesaver.saveAs(new Blob([file], {
          type: 'application/octet-stream'
        }), 'test.zip');
      });
  })
  .catch(err => console.log(err.stack || err));
```

**Comment by ewnd9** [comment]
@efernandesng Do you mind if I merge the pull request as is and add examples by myself?

**Comment by efernandesng** [comment]
@ewnd9 of course not! Sorry, but I had little time.
Thanks

Em 18/09/2016 04:04, "ewnd9" <notifications@github.com> escreveu:

**Pull request merged**